.github/workflows/nightly_build.yml

```diff
@@ -26,24 +27,79 @@ jobs:
         uses: "actions/checkout@v2"
         with:
           ref: ${{ matrix.branch }}
+      - name: "Build if updated during the last 24h"
+        id: updated
+        run: |
+          SHA=$( git rev-parse HEAD )
+          if [ -n "$( git rev-list -n 1 --after="24 hours" $SHA )" ]; then
+            echo "::set-output name=build::yes"
+          else
+            echo "::set-output name=build::no"
+          fi
       - name: "Deploy source into app container"
+        if: ${{ steps.updated.outputs.build == 'yes' }}
         run: |
           sudo cp --no-target-directory --preserve --recursive `pwd` /glpi
           sudo chown -R 1000:1000 /glpi
       - name: "Install dependencies"
+        if: ${{ steps.updated.outputs.build == 'yes' }}
         run: |
           docker exec ${{ job.services.app.id }} composer install --optimize-autoloader --prefer-dist --no-interaction --no-progress --no-suggest
       - name: "Define release name"
+        id: release-name
+        if: ${{ steps.updated.outputs.build == 'yes' }}
         run: |
           REF_NAME=$(echo ${{ matrix.branch }} | sed -E 's|/|-|')
+          echo "::set-output name=base::$REF_NAME"
           SHA=$(git rev-parse --short HEAD)
           echo "release_name=$REF_NAME.$SHA" >> $GITHUB_ENV
       - name: "Build"
+        if: ${{ steps.updated.outputs.build == 'yes' }}
         run: |
           echo "Y" | docker exec --interactive ${{ job.services.app.id }} tools/make_release.sh . ${{ env.release_name }}
           docker cp ${{ job.services.app.id }}:/tmp/glpi-${{ env.release_name }}.tgz ${{ github.workspace }}/${{ env.release_name }}.tar.gz
       - name: "Store archive"
+        if: ${{ steps.updated.outputs.build == 'yes' }}
         uses: actions/upload-artifact@v2
         with:
-            name: ${{ env.release_name }}.tar.gz
+            name: ${{ steps.release-name.outputs.basename}}
             path: ${{ github.workspace }}/${{ env.release_name }}.tar.gz
```

Comment:
Maybe storing the artifact here is useless now. Ping @orthagh

Reply:
IMHO, in the case the following step fails for any reason, this leaves us the possibility to get the artifact and upload it manually on the dedicated repository.

Reply:
Logs and artifacts are stored for 90 days (this is the limit defined for the `glpi-project` org). Maybe we could limit this retention by adding `retention-days: 1` param. I guess that if nightly build upload fails, we will not manually upload more than the last build.

Reply:
Maybe better a week or a month, just in case the fail occurs during a week-end or holidays, or even we just take days to react. Also remember the following build won't occur without a commit during the last 24h so an upload failure won't necessary be fixed with the following nightly build.